### DISCUSSION TITLE
Why does convert use MVG instead of Inkscape?

### DISCUSSION THREAD

**Discussion by TheNextGuy32** [ImageMagick/ImageMagick#1880] [General] [answered]
When I run `convert protests-back.svg protests.png` I get:

    convert: unable to read font `helvetica' @ error/annotate.c/RenderFreetype/1391.
    convert: non-conforming drawing primitive definition `letter-spacing' @ 
    error/draw.c/RenderMVGContent/4434.

This is confusing, as I expect ImageMagick to automatically use my installed Inkscape to open svgs and export them to pngs. Inkscape doesn't have problems with `letter-spacing`, `text`, or `flows`.

I understand that ImageMagick relies on the `delegates.xml` file to detail what programs to use where. Do I need to update `decode="svg"` to specifically use Inkscape? My understanding is that it is failing on the open svg step, not the conversion to png step.

`delegates.xml`:

    <delegate decode="svg" 
    command="&quot;rsvg-convert&quot; -o &quot;%o&quot; &quot;%i&quot;"/>

    

    <delegate decode="svg:decode" stealth="True" 
    command="&quot;inkscape&quot; &quot;%s&quot; 
    --export-png=&quot;%s&quot; 
    --export-dpi=&quot;%s&quot; 
    --export-background=&quot;%s&quot; 
    --export-background-opacity=&quot;%s&quot; &gt; &quot;%s&quot; 2&gt;&amp;1"/>

`convert -list format | grep -i svg` gives:

    MSVG  SVG       rw+   ImageMagick's own SVG internal renderer
    SVG  SVG       rw+   Scalable Vector Graphics (XML 2.9.4)
    SVGZ  SVG       rw+   Compressed Scalable Vector Graphics (XML 2.9.4)

I was able to get it to use Inkscape in the past, or at least something that doesn't give letter-spacing and flow errors, but I reinstalled everything to try to isolate the dependencies of the installation process.

Thanks for any and all help yall!

**Comment by TheNextGuy32** [comment]
The answer was staring at me, since I was using Inkscape 1.0 I have to manually update `export-png` to `export-file`. A bit of an annoying thing to ask my users to do, but it can be done under the hood.

This can be closed.

**Comment by urban-warrior** [comment, marked as answer]
Inkscape changed their command-line options from version 0.92 to 1.0.  In recent releases of ImageMagick, we assume inkscape 1.0 and require a manual change for 0.92 in config/delegates.xml.

**Reply by fmw42** [reply]
I just checked the Inkscape page at https://inkscape.org and version 0.92.4 is still the latest version. Version 1.0 is still in development. Why have you changed ImageMagick for compatibility with Inkscape 1.0 at this time.  Have I misunderstood your comment.

**Comment by fmw42** [comment]
Would you elaborate on the change for what is needed in the delegates.xml file. That might explain why I keep getting an abort trap 6 when converting SVG files via ImageMagick and Inkscape 0.92, even though the output file is created.

**Reply by TheNextGuy32** [reply]
`--export-png=&quot;%s&quot;` -> `--export-file=&quot;%s&quot;`
This is evidently only necessary for those using inkscape 1.0. Which I am.

**Reply by fmw42** [reply]
I was asking for the changes needed for Inkscape 0.92, not 1.0. I assumed that the current delegates.xml file was proper for 1.0 from what urban-warrior said above.

**Comment by urban-warrior** [comment]
List delegates.xml.  There is a comment in the file that tells you exactly what you need to do to support inkscape 0.92 vs. 1.0.

**Comment by fmw42** [comment]
Thank you. That fixed the abort trap 6 issue.

**Comment by mhanberg** [comment]
I am seeing a similar error, as seen here.

```
$ convert -resize 600x300 priv/card.svg card.png
convert: non-conforming drawing primitive definition `image' @ error/draw.c/RenderMVGContent/4452.
```

Running `convert --version` gives me

```
Version: ImageMagick 7.0.10-19 Q16 x86_64 2020-06-12 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib
```

and running `inkscape --version` gives me

```
Inkscape 1.0 (4035a4f, 2020-05-01)
    Pango version: 1.43.0
```

I checked my delegates file, and it looks already set up to use inkscape 1.0.

Any ideas?

Edit:

using the `-verbose` flag gives me

```
'inkscape' '/var/folders/49/9nzqdc7x1xzdq2m9gy0c6pq00000gq/T/magick-28524Ug1yFRDrmA3a' --export-file='/var/folders/49/9nzqdc7x1xzdq2m9gy0c6pq00000gq/T/magick-285249HvP3_NwH4Gr.png' --export-dpi='96' --export-background='rgb(100%,100%,100%)' --export-background-opacity='1' > '/var/folders/49/9nzqdc7x1xzdq2m9gy0c6pq00000gq/T/magick-28524hs1ZkYenOWkZ' 2>&1
mvg:/var/folders/49/9nzqdc7x1xzdq2m9gy0c6pq00000gq/T/magick-285241yTxK8hINNNm=>/var/folders/49/9nzqdc7x1xzdq2m9gy0c6pq00000gq/T/magick-285241yTxK8hINNNm MVG 200x100 200x100+0+0 16-bit sRGB 1879B 0.020u 0:00.010
priv/card.svg MVG 200x100 200x100+0+0 16-bit sRGB 1879B 0.000u 0:00.000
priv/card.svg=>card.png MVG 200x100=>600x300 600x300+0+0 16-bit sRGB 1879B 0.420u 0:00.049
convert: non-conforming drawing primitive definition `image' @ error/draw.c/RenderMVGContent/4452.
```

**Reply by mhanberg** [reply]
It appears that the delegates.xml designates the option `--export-file`, but the correct one is `--export-filename`.

The command now runs without fail (the result is not what i want, but that is a different issue 😅)

**Comment by snibgo** [comment]
From your "-verbose" text, inkscape failed, so IM used MVG instead. Is Inkscape on your system path?

**Reply by mhanberg** [reply]
Yes

```
$ which inkscape
/usr/local/bin/inkscape
```

**Comment by fmw42** [comment]
FYI:  With IM 7, use magick not convert. Also put your -resize after reading the SVG file. IM 7 is more strict about syntax than IM 6.
